src/queries/useChunkedCQLFetch.js

```diff
@@ -0,0 +1,131 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useQueries } from 'react-query';
+
+import { chunk } from 'lodash';
+import useOkapiKy from '../useOkapiKy';
+
+/* When fetching from a potentially large list of items,
+ * make sure to chunk the request to avoid hitting limits.
+
+ * This is a contentious inclusion in stripes-core, as it encourages
+ * frontend joins rather than "proper" API design, and developers should
+ * be aware that this represents something of an antipattern. However, on
+ * occasion frontend joins are the most expedient and efficient ways of completing
+ * cross-module functionality, and so this hook aims to standardise the bulk
+ * fetching of resources from a CQL module.
+ *
+ * See slack thread https://folio-project.slack.com/archives/C210UCHQ9/p1688141053428329
+ * for more context.
+ */
+
+// Only defining defaults to ward of "magic number" sonarlint rule -.-
+const CONCURRENT_REQUESTS_DEFAULT = 5;
```

Comment:
When testing this hook please include one test case with > 300 *unique* IDs (60 * 5).  I could not yet get that scenario to work in ui-inventory as the very last batch of 0<x<=60 IDs would consistently fail, but I was on vacation last week and haven't yet figured out why.

Reply:
Have written basic test coverage, including a test case for that, but since this is unit testing, it's possible something is going wrong with the endpoint returns etc?

Reply:
Certainly possible yes.

Reply:
@maccabeelevine Have you managed to get this working as written yet or is it still not fetching the final page?

Reply:
@EthanFreestone Sorry no, have not had time yet.  It's a very artificial scenario now that the de-duping is in place in ui-inventory so I will have to come up with some more creative test data.  No timeline but I hope soon.

Reply:
@maccabeelevine Apologies for pushing again, but I'd like to merge this whenever possible. Currently not seeing any problems with the equivalent code in `stripes-erm-components`, and would like to merge this if possible.

Could you please try to recreate your issue with it?

Reply:
@EthanFreestone I think you are good to merge it, as you said if there actually is an issue here it is almost definitely on the endpoint side, and the changes made there will have mitigated any real-world example.  Apologies I haven't been able to reproduce yet and won't for a while more, I would not hold up the merge.

Reply:
Thank you Maccabee :) I'll merge asap, hopefully the issue doesn't rear it's head for you down the line.